### ISSUE TITLE
The default session timeout is less than the heartbeat time.

### ISSUE THREAD

**Issue by Cykooz** [aio-libs/sockjs#10]
```python
class SessionManager(dict):
    ...
    def __init__(self, name, app, handler, loop,
                 heartbeat=15.0, timeout=timedelta(seconds=10), debug=False):
```
As a result, when SessionManager._heartbeat_task() is called first time then a session already is expired. When the session is expired, it closes. I think that it is not good :)

I looked at realisation of SockJS for Twisted (https://github.com/DesertBus/sockjs-twisted). In this package option **timeout** is only used to identify the problems during the connection. After the connection is established, this option is not used.

*PS: sorry for my English*

**Comment by fafhrd91** [comment]
that makes sense for websocket connection, but we have polling transports as well.

**Comment by Cykooz** [comment]
So you do not consider this a problem?
I think that this library in the first place should work well with WebSockets, and only then with all the other protocols.

**Comment by fafhrd91** [comment]
timeout should be applied only to non-acquired sessions, so websocket connections should be fine. if we still expire connected websocket connections then it is bug

**Comment by Cykooz** [comment]
```python
    def _heartbeat_task(self):
        ...
                if session.expires < now:
                    # Session is to be GC'd immedietely
                    if session.id in self.acquired:
                        yield from self.release(session)
                    if session.state == STATE_OPEN:
                        yield from session._remote_close()
                    if session.state == STATE_CLOSING:
                        yield from session._remote_closed()

                    del self[session.id]
                    del self.sessions[idx]
                    continue

                elif session.acquired:
                    session._heartbeat()
```
If I understand this code, it does not matter what the session acquired or no. If the session is expired, it will be deleted. And it is this behavior I see in my application.

**Comment by fafhrd91** [comment]
i think websocket transport should do heartbeating. in my old implementation `https://github.com/fafhrd91/pyramid_sockjs/blob/master/pyramid_sockjs/transports/websocket.py#L46` i used heartbeating inside transport.

**Comment by Cykooz** [comment]
And how heartbeating inside transport will help with the session expiration if  session timeout is less than heartbeat timeout?

Maybe just swap conditions? Like this:
```python
                if session.acquired:
                    session._heartbeat()

                elif session.expires < now:
                    ...
                    del self[session.id]
                    del self.sessions[idx]
                    continue
```

**Comment by fafhrd91** [comment]
we can not relay on session.acquired, http is just broken.
really we only have problem with websockets, and it is easier to solve on transport level. basically we just need to call session._tick() from websocket while it is connected.

**Comment by Cykooz** [comment]
So.. Then we need to call session._tick() in further, and faster than the session timeout will finished.

**Comment by Cykooz** [comment]
Or... if fully remove the heartbeat from "SessionManager" then it may help.

**Comment by fafhrd91** [comment]
i suppose this should bug fixed with #11?

**Comment by Cykooz** [comment]
Yes it is, this bug fixed.
